process/block/preprocess/smartContractResults.go

```diff
@@ -510,23 +524,38 @@ func (scr *smartContractResults) ProcessMiniBlock(miniBlock *block.MiniBlock, ha
 	}
 	maxGasLimitUsedForDestMeTxs := scr.economicsFee.MaxGasLimitPerBlock(scr.shardCoordinator.SelfId()) * maxGasLimitPercentUsedForDestMeTxs / 100
 
-	log.Trace("smartContractResults.ProcessMiniBlock", "totalGasConsumedInSelfShard", gasInfo.totalGasConsumedInSelfShard)
+	log.Debug("smartContractResults.ProcessMiniBlock: before processing",
+		"totalGasConsumedInSelfShard", gasInfo.totalGasConsumedInSelfShard,
+		"total gas provided", scr.gasHandler.TotalGasProvided(),
+		"total gas refunded", scr.gasHandler.TotalGasRefunded(),
+		"total gas penalized", scr.gasHandler.TotalGasPenalized(),
+	)
+	defer func() {
+		log.Debug("smartContractResults.ProcessMiniBlock after processing",
+			"totalGasConsumedInSelfShard", gasInfo.totalGasConsumedInSelfShard,
+			"gasConsumedByMiniBlockInReceiverShard", gasInfo.gasConsumedByMiniBlockInReceiverShard,
+			"num scrs processed", numSCRsProcessed,
+			"total gas provided", scr.gasHandler.TotalGasProvided(),
+			"total gas refunded", scr.gasHandler.TotalGasRefunded(),
+			"total gas penalized", scr.gasHandler.TotalGasPenalized(),
+		)
+	}()
 
 	for index := range miniBlockScrs {
 		if !haveTime() {
 			err = process.ErrTimeIsOut
 			return processedTxHashes, index, err
 		}
 
-		gasProvidedByTxInSelfShard, errComputeGas := scr.computeGasProvided(
+		gasProvidedByTxInSelfShard, err = scr.computeGasProvided(
```

Comment:
This is a main reason that using defer function that handles an internal err variable is a bad practice. 
This should have been split in at least 3 functions:

```go
scr.processSomethingWithoutErrHandling()
err := scr.processOtherStuffWithErrHandling()
scr.handleErr(err)
```

In this way the intent will be better shown and the code would have been more robust

Reply:
:+1: